java/src/main/java/org/whispersystems/libsignal/SignalProtocolAddress.java

```diff
@@ -7,25 +7,38 @@
 
 public class SignalProtocolAddress {
 
-  private final String name;
-  private final int    deviceId;
+  private static native long New(String name, int device_id);
```

Comment:
Is there something requiring these to be uppercase? Reading something like `return Name(blah)` looks like a compiler error.

Reply:
No it is not required. I just wanted something that distinguished these native calls vs the Java code, especially as in some cases they are the same name otherwise (eg doFoo calling native DoFoo). I'm not that familiar with Java idioms so it seemed ok to me but if it looks awkward to you it is probably wrong. Would something like `nativeNew` etc be better?

Reply:
`nativeNew` would be better. CamelCase is generally reserved for types. HOWEVER, if these are all internal (not exposed to API consumers) then I don't really mind the usage here. Others might have grievances though.

Reply:
A quick Python script later, https://github.com/signalapp/libsignal-jni/pull/4 adds `native` prefix on the JNI bindings and I've already pushed a commit changing this PR to use same

Reply:
Thankfully you could script it instead of doing that by hand :+1: